### PR TITLE
Introduce jet cache manager

### LINKED ISSUE #483: Startup fails in a Spring boot application.
I am trying to start Jet as a simple spring boot application with no other dependencies.

Added a Bean that creates a Jet instance with default configuration.

>  @Bean
    public JetInstance getJetInstance(){
        return Jet.newJetInstance();
    }

Jet version is 0.4

Below is the error:
Caused by: java.lang.IllegalStateException: Failed to introspect Class [com.hazelcast.jet.impl.JetInstanceImpl] from ClassLoader [sun.misc.Launcher$AppClassLoader@18b4aac2]
	at org.springframework.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:656) ~[spring-core-5.0.0.RC2.jar:5.0.0.RC2]

Caused by: java.lang.ClassNotFoundException: javax.cache.Cache
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381) ~[na:1.8.0_74]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424) ~[na:1.8.0_74]
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:331) ~[na:1.8.0_74]
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357) ~[na:1.8.0_74]
	... 47 common frames omitted

### PR DESCRIPTION
**Pull request by gurbuzali**
fixes #483

### CODE REVIEW

File: hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetCacheManagerImpl.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    @Override
+    public <K, V> IStreamCache<K, V> getCache(String name) {
+        return CacheGetter.getCache(jetInstance, name);
+    }
+
+    /**
+     * This class is necessary to conceal the cache-api
+     */
+    private static class CacheGetter {
```
Comment (cangencer) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetCacheManagerImpl.java]: is this class still necessary?

Comment (gurbuzali) [inline hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JetCacheManagerImpl.java]: yes, it is still necessary

**Comment by cangencer** [comment]
code samples will also need updating after merging.

**Comment by gurbuzali** [comment]
Code samples change https://github.com/hazelcast/hazelcast-jet-code-samples/pull/49

**Pull request merged**